runtime/src/migrations.rs

```diff
@@ -83,3 +86,22 @@ impl OnRuntimeUpgrade for GrantsMigration {
         weight
     }
 }
+
+#[cfg(feature = "with-staking")]
+pub struct StakingMigration;
+
+#[cfg(feature = "with-staking")]
+impl OnRuntimeUpgrade for StakingMigration {
+    fn on_runtime_upgrade() -> Weight {
+        let mut weight = Weight::from(0u64);
+
+        debug::RuntimeLogger::init();
+        debug::print!("🕊️ Starting staking migration...");
+
+        weight = weight.saturating_add(NodleStaking::on_runtime_upgrade());
+
+        debug::print!("🕊️ Staking migration Done");
+
+        weight
+    }
+}
```

Comment:
I don't think we need to call the on_runtime_upgrade function? Should be called by substrate already

Reply:
Nice one, removed the messy stuff  Done :+1: